### ISSUE TITLE
Mouse Pointer Stuck to Top Left in Drag Preview Despite Using `preserveOffsetOnSource` Utility

### ISSUE THREAD

**Issue by RabiRoshan** [atlassian/pragmatic-drag-and-drop#54]
### Issue:

I'm attempting to replicate the [Board Example](https://atlassian.design/components/pragmatic-drag-and-drop/examples/#board), but I'm encountering an issue where the mouse pointer on the drag preview is consistently stuck to the top left corner, unlike the example in the documentation. I'm unsure if I'm misinterpreting the utility `preserveOffsetOnSource`, but it doesn't seem to work for me. In the meantime, I found a workaround using the solution mentioned here: https://github.com/atlassian/pragmatic-drag-and-drop/issues/25#issuecomment-2043678626. It would be fantastic if this issue could be resolved according to the documentation.

This library is excellent, and I would love to contribute! ❤️

### Versions:
- `"react": "^18.2.0"`
- `"@atlaskit/pragmatic-drag-and-drop": "^1.1.7"`

### Import used:
```javascript
import { preserveOffsetOnSource } from '@atlaskit/pragmatic-drag-and-drop/element/preserve-offset-on-source'
```

**Comment by alexreardon** [comment]
Hi there,

Could you please post a codesandbox with the issue you are experiencing? It's difficult to know what is going on without an example to play with.

**Comment by RabiRoshan** [comment]
Hey @alexreardon,
here is the codesandbox as requested: https://codesandbox.io/p/sandbox/pragmatic-dnd-preserveoffsetonsource-kr2cz2

**Comment by alexreardon** [comment]
Thank you for the example. It's really helpful. Taking a look now

**Comment by RabiRoshan** [comment]
Hey @alexreardon , just dropping a reminder 🙌.

**Comment by alexreardon** [comment]
Hi there,

Sorry for the delay on this one. A fix has now shipped in version `1.2.3`.

Cheers

**Comment by nmhungdev** [comment]
@alexreardon 
How to archive `preserveOffsetOnSource` with [non-native-custom-drag-previews](https://atlassian.design/components/pragmatic-drag-and-drop/core-package/adapters/element/drag-previews#non-native-custom-drag-previews)
[Sandbox](https://codesandbox.io/p/sandbox/pragmatic-dnd-preserveoffsetonsource-kr2cz2)
Do you have any suggestions? I would greatly appreciate them.
